### ISSUE TITLE
Add Documentation for use with `go build`

### ISSUE THREAD

**Issue by Naatan** [wailsapp/wails#691]
**Is your feature request related to a problem? Please describe.**

Currently all the documentation focuses on `wails build`, which is great if that happens to work well with your workflow, and not so great if it doesn't. 

From scanning the debug output of `wails build` I was able to figure out how to build it directly with `go build`, so it's perfectible feasible it would seem. I'm a little confused why you opted for your own build system, at the very least you should cover the build system that your audience is already familiar with.

**Describe the solution you'd like**

Cover `go build` use-cases. For example it's not evident how to debug a wails app without using the `wails build` command.

**Describe alternatives you've considered**

None. Using `wails build` simply isn't viable for our use-case.

**Comment by leaanthony** [comment]
There's a fair bit of pre-processing that goes on before a build can happen and that's why there's a build utility. Recently we moved to using go's native embed instead of our own solution, which means some of that preprocessing goes away. Out of interest, what's the drivers for you wanting to do this manually? Is it mostly for debugging bad builds? I can certainly look at adding docs for it, but there's a lot of higher priority tickets right now so it might take a while.

**Comment by Naatan** [comment]
Basically the wails build system locks you into too many things:

- You are required to have a go.mod at the root of your project (not friendly to mono repos)
- You are required to have a wails project json (we already have build tooling for this, don't want yet another json file)
- You are required to run custom tooling (again, already have build tooling)

Additionally I don't see why the wails build system needs to concern itself with the frontend. Great to have it as an option I suppose, but it feels unnecessary.

Basically the `wails` cmd line seems like is friendly optional tooling if you want to buy into the workflow it enforces, but it should be optional.

Mind you I've been able to work around the `wails` command entirely just by looking at what it does. So it seems perfectly possible, and just needs a little documentation.

It really all comes down to this though; if you document the low level usage then you enable many more workflows than just the one you are concerning yourself with through the `wails` command.

**Comment by leaanthony** [comment]
I don't suppose you took notes? It seems that you've already got all the information that someone else with the same concerns would need. Happy to accept a PR!

**Comment by Naatan** [comment]
I can see if I have some time this weekend, but can't promise anything. Either way I likely wouldn't be able to cover all relevant spots in the docs as I am not that familiar.

For what it's worth, this is the build command I'm using:

```
go build -o <outputfile> -ldflags "-w -X github.com/wailsapp/wails.BuildMode=debug" <input>
```

It's pretty much just the ldflags, if I build without those it doesn't produce a usable executable. For production we'd build it with `wails.BuildMode =prod` instead of `debug`.

Those same flags also work with `go run`.

As for the frontend, we build that with our own tooling, wails doesn't need to be aware of that.

If you're curious you can see the PR that adds it to our project here: https://github.com/ActiveState/cli/pull/1320

That's still pretty rough though, we're at the PoC stage.

**Comment by leaanthony** [comment]
Hey that's great! I used to use ActiveState Python back in the day so this is slightly mind blowing for me 🤯 I'll look at filling in the gaps in the docs if you do a PR. It's useful feedback for v2. I'm not sure it'll be possible to not use the CLI at all as we do generate some C headers but potentially I can add a flag/tool for doing just that part. Appreciate the constructive feedback 👍

**Comment by leaanthony** [comment]
Based on your feedback, for v2, we'll output the commands being run when the verbose flag is set.

**Comment by Naatan** [comment]
So I didn't have time this weekend, but that turns out to be a good thing. Today I started working on the Windows implementation of our use-case for wails and found that on Windows the build flags are different. Additionally there's a bunch of special know-how to cover regarding the debugging of wails apps on Windows that would not be self-evident without the `wails` command.

I think if I were to PR a docs update that update would likely cause more issues than it would fix. I suggest instead maybe putting a note that you currently simply do not support building without the `wails` command, but that you can have mixed success by copying the commands from the `wails build -v` command output. 

Additionally while wails is a great library I think what I am coming around to is that it's only really useful if you want a comprehensive solution. For low level implementations like ours it's likely better to use the webview package directly, maybe that's also something the docs could cover.

None of this is a criticism of wails, I think you're doing great covering your intended use-case. This is more me coming to the realization that wails does a bit too much for our use-case.

**Comment by leaanthony** [comment]
Thanks for the feedback. You are right: the aim of the project is to simplify development using these technologies as there are a bunch of hard problems to solve if you are using just the core webview library. For some people, that's what they want to do so I think you've made the right choice. Regarding debugging on Windows, there is a guide here: https://wails.app/guides/debugging/#windows

Good luck with your endeavours!
